packages/active-campaign-client/src/helpers/getUserFromCognito.ts

```diff
@@ -4,9 +4,15 @@ import { QueueEvent } from '../types/queueEvent';
 import { listUsersCommandOutputToUser } from './listUsersCommandOutputToUser';
 
 export async function getUserFromCognito(queueEvent: QueueEvent) {
+  return await getUserFromCognitoByUsername(
+    queueEvent.detail.additionalEventData.sub
+  );
+}
+
+export async function getUserFromCognitoByUsername(username: string) {
   const command = new ListUsersCommand({
     UserPoolId: process.env.COGNITO_USER_POOL_ID,
-    Filter: `username = "${queueEvent.detail.additionalEventData.sub}"`,
+    Filter: `username = "${username}"`,
```

Comment:
This could be updated this way to avoid in `resyncUser` the empty try catch:
```
export async function getUserFromCognito(queueEvent: QueueEvent) {
  const username = queueEvent.detail.additionalEventData.sub;
  const user = await getUserFromCognitoByUsername(username);
  if (!user) {
    // eslint-disable-next-line functional/no-throw-statements
    throw new Error('User not found');
  }
  return user;
}

export async function getUserFromCognitoByUsername(username: string) {
  const command = new ListUsersCommand({
    UserPoolId: process.env.COGNITO_USER_POOL_ID,
    Filter: `username = "${username}"`,
  });
  const listUsersCommandOutput = await cognitoClient.send(command);
  const user = listUsersCommandOutputToUser(listUsersCommandOutput);
  console.log('User:', JSON.stringify(user, null, 2)); // TODO: Remove after testing
  return user;
}

```